backend/API/main.py

```diff
@@ -0,0 +1,109 @@
+import os
+from typing import Any
+from flask import Blueprint, render_template, Response
+from flask.globals import request, session
+from flask.helpers import flash, send_file
+from flask_login import login_required, current_user
+from . import db, upload_folder
+from .models import User, CAFF, Comment
+
+main = Blueprint('main', __name__)
+
+@main.route('/caffs', methods=['GET'])
+@login_required
+def listCAFFs():
+    #TODO good arg parse maybe good
+    filter=request.args.get("filter")
+    caffs = db.session.query(CAFF).all()
+    json_data = []
+    for caff in caffs:
+        if (not filter) or (filter in caff.name):
+            tmp = {"id" : caff.id,
+                    "name" : caff.name,
+                    "number_of_comments" : caff.comments}
+            json_data.append(tmp.copy())
+    return {"caffs" : json_data}
+
+@main.route('/caffs', methods=['POST'])
+@login_required
+def upload():
+    #prolly works
+    file= request.files['file']
+    caff = CAFF.query.filter_by(name=file.filename, uploader=current_user.id).first()
+    if caff or file.filename.split('.')[-1] != "caff":
+        return Response(response='File already uploaded', status=400)
+    new_caff = CAFF(url=upload_folder+"\\"+file.filename, name=file.filename, comments =0, uploader=current_user.id)
```

Comment:
upload_folder already contains the path separator (`\`)

```suggestion
    new_caff = CAFF(url=upload_folder+file.filename, name=file.filename, comments =0, uploader=current_user.id)
```